pkg/util/helper/workstatus.go

```diff
@@ -50,10 +50,11 @@ func AggregateResourceBindingWorkStatus(c client.Client, binding *workv1alpha2.R
 		return err
 	}
 
+	fullyAppliedCondition := generateFullyAppliedCondition(binding.Spec, aggregatedStatuses)
+
 	currentBindingStatus := binding.Status.DeepCopy()
 	currentBindingStatus.AggregatedStatus = aggregatedStatuses
-	meta.SetStatusCondition(&currentBindingStatus.Conditions, generateFullyAppliedCondition(binding.Spec, aggregatedStatuses))
-
+	meta.SetStatusCondition(&currentBindingStatus.Conditions, fullyAppliedCondition)
```

Comment:
I think it'd be better to set the condition inside of `retry.RetryOnConflict`, just like what we did at [execution controller](https://github.com/karmada-io/karmada/blob/95c6c0971b2b3d5edfdb1af2caf5e043d2b10442/pkg/controllers/execution/execution_controller.go#L267-L268).

You can move the `DeepEqual` too.

Reply:
Updated.